src/main/flight/pid_init.c

```diff
@@ -428,9 +428,11 @@ void pidInitConfig(const pidProfile_t *pidProfile)
     pidRuntime.tpaBreakpoint = constrainf((pidProfile->tpa_breakpoint - PWM_RANGE_MIN) / 1000.0f, 0.0f, 0.99f);
     // default of 1350 returns 0.35. range limited to 0 to 0.99
     pidRuntime.tpaMultiplier = (pidProfile->tpa_rate / 100.0f) / (1.0f - pidRuntime.tpaBreakpoint);
-    // it is assumed that tpaLowBreakpoint is always less than or equal to tpaBreakpoint
-    pidRuntime.tpaLowBreakpoint = constrainf((pidProfile->tpa_low_breakpoint - PWM_RANGE_MIN) / 1000.0f, 0.01f, 1.0f);
+    pidRuntime.tpaLowBreakpoint = constrainf((pidProfile->tpa_low_breakpoint - PWM_RANGE_MIN) / 1000.0f, 0.0f, 1.0f);
+    // tpaLowBreakpoint must be less than or equal to tpaBreakpoint
     pidRuntime.tpaLowBreakpoint = MIN(pidRuntime.tpaLowBreakpoint, pidRuntime.tpaBreakpoint);
+    // disable TPA Low if its breakpoint is 1000 (or lower), or if TPA itself is disabled by setting the tpa_breakpoint to 1000
+    pidRuntime.tpaLowDisabled = pidRuntime.tpaLowBreakpoint <= 0.01f;
     pidRuntime.tpaLowMultiplier = pidProfile->tpa_low_rate / (100.0f * pidRuntime.tpaLowBreakpoint);
     pidRuntime.tpaLowAlways = pidProfile->tpa_low_always;
```

Comment:
```suggestion
    if (pidProfile->tpa_low_breakpoint > PWM_RANGE_MIN) {
        // tpaLowBreakpoint must be less than or equal to tpaBreakpoint
        pidRuntime.tpaLowBreakpoint = constrainf((pidProfile->tpa_low_breakpoint - PWM_RANGE_MIN) / 1000.0f, 0.0f, pidRuntime.tpaBreakpoint);
        pidRuntime.tpaLowDisabled = false;
        pidRuntime.tpaLowMultiplier = pidProfile->tpa_low_rate / (100.0f * pidRuntime.tpaLowBreakpoint);
        pidRuntime.tpaLowAlways = pidProfile->tpa_low_always;
    } else {
       pidRuntime.tpaLowDisabled = true;
       pidRuntime.tpaLowBreakpoint = 0.0f;
   }
```

There was possible division by zero in `pidProfile->tpa_low_rate / (100.0f * pidRuntime.tpaLowBreakpoint);`.

Reply:
Ah, stupid of me to not see that.  Thank you!

Your rewrite is very nice.

However, in your suggestion, we no longer have the minimum `pidRuntime.tpaLowBreakpoint` of 0.01, caused by setting `tpaLowDisabled` false below 0.01.  

I think not allowing a breakpoint below was intentional, simply because there was not much point going any lower than some 'reasonable' min value, the slope of the line becomes too steep, and throttle noise could trigger it off and on abruptly.  Hence perhaps to maintain the current functionality, the opening conditional could be `if (pidProfile->tpa_low_breakpoint > PWM_RANGE_MIN + 10) {`, and the two instances of `0.0f` could be `0.01f`?

Also, could I ask, although we only use `pidRuntime.tpaLowAlways` in the first situation, is it good practice to set it anyway (regardless of whether it is needed?)

Reply:
`pidRuntime.tpaLowBreakpoint` of 0.01 - One thing this did is to avoid division by zero.

> I think not allowing a breakpoint below was intentional, simply because there was not much point going any lower than some 'reasonable' min value, the slope of the line becomes too steep, and throttle noise could trigger it off and on abruptly. Hence perhaps to maintain the current functionality, the opening conditional could be if (pidProfile->tpa_low_breakpoint > PWM_RANGE_MIN + 10) {, and the two instances of 0.0f could be 0.01f?

I don't  know. Nothing should overflow with tpaLowBreakpoint=1001, and 1% seems pretty arbitrary, so I'll let user decide (but there may be consequences). Or define `TPA_LOW_MIN` (either 10 or 1010). No need to use 0.01f, `if` will handle minimum.

Reply:
`pidRuntime.tpaLowAlways = false; pidRuntime.tpaLowMultiplier = 0; ` is good idea

Reply:
I think the problem is a practical one, of abrupt changes in the attenuated parameters (P, D) when throttle comes close to zero if the transition range is very short.  That may cause a 'glitch' in the motor outputs due to the abrupt loss of PIDSum just because throttle was low.  In practice it would be inadvisable to use this much below 5% I think.